pygmt/tests/test_grdimage_image.py

```diff
@@ -19,26 +15,8 @@ def fixture_xr_image():
     """
     geotiff = which(fname="@earth_day_01d_p", download="c")
     with rioxarray.open_rasterio(filename=geotiff) as rda:
-        if len(rda.band) == 1:
-            with rasterio.open(fp=geotiff) as src:
-                df_colormap = pd.DataFrame.from_dict(
-                    data=src.colormap(1), orient="index"
-                )
-                array = src.read()
-
-                red = np.vectorize(df_colormap[0].get)(array)
-                green = np.vectorize(df_colormap[1].get)(array)
-                blue = np.vectorize(df_colormap[2].get)(array)
-                # alpha = np.vectorize(df_colormap[3].get)(array)
-
-            rda.data = red
-            da_red = rda.astype(dtype=np.uint8).copy()
-            rda.data = green
-            da_green = rda.astype(dtype=np.uint8).copy()
-            rda.data = blue
-            da_blue = rda.astype(dtype=np.uint8).copy()
-
-            xr_image = xr.concat(objs=[da_red, da_green, da_blue], dim="band")
+        if len(rda.band) == 3:
+            xr_image = rda.load()
```

Comment:
Wait for https://github.com/GenericMappingTools/gmtserver-admin/issues/257, to see if the 3-band GeoTIFF file stays, or we revert back to the quantized/indexed GeoTIFF file from before Sep 2023.

Reply:
Gonna assume that the 3-band GeoTIFF will stay for now, so that we can move on with the GMT 6.5.0 update. Marked this PR as ready for review.

Reply:
What about splitting this PR into two separate RPs so if the upstream decide to to back to the indexed geotiff file, we can easily revert the changes.

Reply:
I kinda want to take the code from this fixture and just make a `load_blue_marble` dataset function after this :slightly_smiling_face: If we do go back to the indexed GeoTIFF files, then the modifications would go in that `load_blue_marble` function.